includes/data-stores/class-wc-customer-data-store.php

```diff
@@ -0,0 +1,242 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC Customer Data Store.
+ *
+ * @version  2.7.0
+ * @category Class
+ * @author   WooThemes
+ */
+class WC_Customer_Data_Store implements WC_Customer_Data_Store_Interface, WC_Object_Data_Store {
+
+	/**
+	 * Method to create a new customer in the database.
+	 *
+	 * @since 2.7.0
+	 * @param WC_Customer
+	 */
+	public function create( &$customer ) {
+		$customer_id = wc_create_new_customer( $customer->get_email(), $customer->get_username(), $customer->get_password() );
+
+		if ( ! is_wp_error( $customer_id ) ) {
+			$customer->set_id( $customer_id );
+			$this->update_user_meta( $customer );
+			wp_update_user( array( 'ID' => $customer->get_id(), 'role' => $customer->get_role() ) );
+			$wp_user = new WP_User( $customer->get_id() );
+			$customer->set_date_created( strtotime( $wp_user->user_registered ) );
+			$customer->set_date_modified( get_user_meta( $customer->get_id(), 'last_update', true ) );
+			$customer->read_meta_data();
```

Comment:
I think save_meta_data needs called in `create` just like update. https://github.com/woocommerce/woocommerce/pull/12353/files#diff-d6dba1e5e24baa6d5d15671305dd1994R85

I think we also want to call `apply_changes` here and in update so the changes array is reset. I just realized this is missing from coupons too, so I'll add it there.